### PR TITLE
feat(test): default ry test -p workers to cores minus one

### PR DESCRIPTION
**Pull request by t0k0sh1**
## Summary

- `ry test -p` (no explicit count) now uses `hardware_concurrency() - 1` (min 1) so one core is left free for the rest of the system. `hw == 0` (unknown) or `hw == 1` both yield 1 worker. Explicit `-p N` and sequential `ry test` are unaffected.
- Adds a pure helper `computeDefaultWorkers(unsigned)` exposed in `include/ry/jit/test_runner.hpp` so the boundary cases (`hw == 0/1/2/UINT_MAX`) are unit-testable without depending on the host CPU count, and the unsigned-underflow guard is locked in.
- `runTestFilesParallel` now prints `Running M test files with K workers...` to stderr at the start of a parallel run so the active worker count is visible from the console as execution begins (the end-of-run `(K workers)` summary stays).

Closes #2216

## Test plan

- [x] `./build-rust/ry_tests --gtest_filter='ComputeParallelism.*:ComputeDefaultWorkers.*'` — 18/18 pass
- [x] `./build-rust/ry test -p tests/spec` — 207/207 pass, opening line shows `Running 207 test files with 9 workers...` (hw.ncpu = 10)
- [x] `./build-rust/ry test -p 2 tests/spec` / `--parallel=3` — opening line reflects the explicit count
- [x] ASan + UBSan via `./.claude/skills/pre-commit-checklist/run-asan.sh` — 207/207 pass
- [x] TSan via `./.claude/skills/pre-commit-checklist/run-tsan.sh` — 207/207 pass
- [x] `./.claude/skills/pre-commit-checklist/run-static-analysis-all.sh` — clean for the diff (the 2 advisory `scan-build` warnings touch pre-existing code outside this change)

## Release Notes

* **Bug Fixes**
  * Default parallel test execution now uses **CPU count − 1** with a **minimum of 1** when running `ry test -p/--parallel` without an explicit worker number.

* **New Features**
  * Parallel runs now print an additional startup line showing the **number of test files** and **worker threads** being used.

* **Documentation**
  * Updated CLI help and reference docs to match the new default worker-count behavior.

* **Tests**
  * Expanded unit tests to cover the parallelism and default worker calculations.

### CODE REVIEW

File: include/ry/jit/test_runner.hpp [hunk 1]
```diff
@@ -7,9 +7,16 @@ namespace ry {
...
+// core free, with a floor of 1 (hw == 0 / hw == 1 both return 1, also guarding
+// against unsigned underflow at hw == 0).
+unsigned computeDefaultWorkers(unsigned hw_concurrency);
+
 // Returns the effective worker count for parallel test execution.
-// requested_workers == 0 means "use std::thread::hardware_concurrency()".
-// The result is clamped to test_file_count (when non-zero) and floored at 1.
+// requested_workers == 0 means "use computeDefaultWorkers(hardware_concurrency())"
+// (i.e. hw - 1, minimum 1). The result is clamped to test_file_count
+// (when non-zero) and floored at 1.
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply include/ry/jit/test_runner.hpp]: Good catch — the implementation takes the default path for `requested_workers <= 0` (the `if (requested_workers > 0)` branch in `src/jit/test_runner.cpp`). Widened the header contract to `<= 0` so the documented behavior matches reality. (In practice the CLI rejects zero/negative explicitly via `parseWorkerCount`, but the header is the public unit-test surface, so the contract should describe the function as it actually behaves.)
